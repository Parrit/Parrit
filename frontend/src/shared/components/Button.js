"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Button = void 0;
var react_1 = __importDefault(require("react"));
var Button = function (props) {
    var type = props.type || "button";
    return (react_1.default.createElement("button", { type: type, title: props.tooltip, className: props.className, onClick: props.clickFunction },
        react_1.default.createElement("span", { className: "name" }, props.name),
        react_1.default.createElement("span", { className: "short-name" }, props.shortName || props.name)));
};
exports.Button = Button;
