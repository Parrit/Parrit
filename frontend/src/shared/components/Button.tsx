import React from "react";

interface Props {
  name: string;
  shortName?: string;
  className?: string;
  clickFunction?: VoidFunction;
  type?: "button" | "submit" | "reset" | undefined;
  tooltip?: string;
}

export const Button: React.FC<Props> = (props) => {
  const type = props.type || "button";

  return (
    <button
      type={type}
      title={props.tooltip}
      className={props.className}
      onClick={props.clickFunction}
      aria-label={props.name}
    >
      <span className="name">{props.name}</span>
      <span className="short-name">{props.shortName || props.name}</span>
    </button>
  );
};
