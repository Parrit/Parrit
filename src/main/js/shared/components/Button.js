var React = require('react');

var Button = React.createClass({
    propTypes: {
        name: React.PropTypes.string.isRequired,
        shortName: React.PropTypes.string,
        className: React.PropTypes.string,
        clickFunction: React.PropTypes.func,
        type: React.PropTypes.string,
		tooltip: React.PropTypes.string
    },

    render: function () {
        var type = this.props.type || 'button';
        return <button type={type} title={this.props.tooltip} className={this.props.className} onClick={this.props.clickFunction}>
            <span className="name">{this.props.name}</span>
            <span className="short-name">{this.props.shortName || this.props.name}</span>
        </button>;
    }
});

module.exports = Button;

