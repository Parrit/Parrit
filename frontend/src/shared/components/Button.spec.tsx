import * as React from 'react'
import * as renderer from 'react-test-renderer';
import {Button} from "./Button";
import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe('<Button/>', () => {
    let props;

    beforeEach(() => {
        props = {
            name: 'button',
            className: 'ooga-booga'
        }
    })

    it('should render correctly', () => {
        const tree = renderer.create(<Button name='button' className='ooga-booga'/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should use the passed in button type and short name if provided', () => {
        props.type = 'submit'
        props.shortName = 'short'
        const tree = renderer.create(<Button {...props} />)
        expect(tree).toMatchSnapshot();
    })

    it('should call the click function when clicked if provided', () => {
        props.clickFunction = jest.fn();
        render(<Button {...props} />)

        userEvent.click(screen.getByLabelText(props.name))
        expect(props.clickFunction).toHaveBeenCalled()
    })
})