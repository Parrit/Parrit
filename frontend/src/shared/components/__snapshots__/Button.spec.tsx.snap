// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render correctly 1`] = `
<button
  aria-label="button"
  className="ooga-booga"
  type="button"
>
  <span
    className="name"
  >
    button
  </span>
  <span
    className="short-name"
  >
    button
  </span>
</button>
`;

exports[`<Button/> should use the passed in button type and short name if provided 1`] = `
<button
  aria-label="button"
  className="ooga-booga"
  type="submit"
>
  <span
    className="name"
  >
    button
  </span>
  <span
    className="short-name"
  >
    short
  </span>
</button>
`;
